### DISCUSSION TITLE
Is it possible to use a custom sqlite build?

### DISCUSSION THREAD

**Discussion by kulshekhar** [prisma/prisma#4108] [Q&A] [answered]
There are cases when a custom sqlite build/amalgamation needs to be used when installing and using node-sqlite3.

Is something like this possible with prisma when it's used with sqlite?

**Comment by ryands17** [comment, marked as answer]
Hey @kulshekhar 👋 
Currently SQLite is compiled to the Prisma Engine binaries https://www.prisma.io/docs/more/supported-databases.

So if you want to use your own custom sqlite with Prisma, you would need to compile it by yourself, changing the compilation settings.

Let me know if that helps :)

**Comment by kulshekhar** [comment]
That does help. Thanks Ryan!

**Reply by ryands17** [reply]
Hey @kulshekhar 👋 
Did that work? There's a doc [here](https://github.com/prisma/prisma-engines#building-prisma-engines) for building Prisma engines if you're stuck someplace.

**Reply by kulshekhar** [reply]
@ryands17 I haven't yet tried that but plan to do so over the weekend. I'll post an update here after I do that

**Reply by kulshekhar** [reply]
I just tried to do this. The build succeeded but it did not pick the custom sqlite version. It's quite likely I did something wrong here.

I assumed setting the `SQLITE3_LIB_DIR` env variable to the custom sqlite directory would be sufficient but it wasn't. This in turn was based on the assumption that the sqlite related parts are handled by rusqlite which uses this env var to build against a custom sqlite version (https://github.com/rusqlite/rusqlite#notes-on-building-rusqlite-and-libsqlite3-sys)

I'll have to dig into this further to figure out how to get this done.

**Reply by shinr3x** [reply]
@kulshekhar, did you manage to find a way of using a custom sqlite build?
I am trying to add sqlcipher functionality to prisma.

**Reply by kulshekhar** [reply]
@DragosCocirlea I was unable to. The post above summarizes my last attempt with this

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
